examples/run_parallel_and_cleanup.py

```diff
@@ -0,0 +1,86 @@
+"""Script to run all example files"""
+import os
+import subprocess
+import shutil
+import sys
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+#checks if we are in the examples directory
+if not os.path.basename(os.getcwd()) == "examples":
+    print("Run this script from the examples directory.")
+    sys.exit(1)
+# Parse command-line arguments
+if len(sys.argv) > 2:
+    print("Usage: python run_parallel_and_cleanup.py [max_threads]")
+    sys.exit(1)
+
+# Default to 2 threads if no argument is provided
+max_threads = int(sys.argv[1]) if len(sys.argv) == 2 else 4
+if max_threads <= 0:
+    print("Maximum number of threads must be a positive integer.")
+    sys.exit(1)
+
+# Define the log directory
+log_dir = "logs"
+os.makedirs(log_dir, exist_ok=True)
+
+
+# Function to run a Python script and log its output
+def run_script(file_path):
+    script_name = os.path.basename(file_path)
+    log_file = os.path.join(log_dir, f"{script_name}.log")
+    print(f"Running {file_path}...")
+
+    try:
+        # Run the script and capture stdout/stderr
+        result = subprocess.run(["python3", file_path],
+                                capture_output=True,
+                                text=True,
+                                check=True)
+        # Write logs to the log file
```

Comment:
https://inductiva.ai/blog/article/programming-at-inductiva#dry-dont-repeat-yourself

Reply:
this is the part that can be quite consolidated, because the block below is almost the same

        with open(log_file, "w", encoding="utf-8") as log:
            log.write(f"--- Output for {script_name} ---\n")
            log.write("STDOUT:\n")
            log.write(result.stdout + "\n")
            log.write("STDERR:\n")
            log.write(result.stderr + "\n")
        print(f"Logs for {script_name} saved to {log_file}")

Reply:
I changed things a bit. We still have the try except but now we only have one time this part that writes to the file